bentoml/yatai/repository/abs_repository.py

```diff
@@ -0,0 +1,151 @@
+# Copyright 2019 Atalaya Tech, Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from datetime import datetime, timedelta
+import os
+from urllib.parse import urlparse
+from dependency_injector.wiring import Provide, inject
```

Comment:
We may need to change this import after PR https://github.com/bentoml/BentoML/pull/1697